sled-agent/tests/integration_tests/early_network.rs

```diff
@@ -0,0 +1,154 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Tests that EarlyNetworkConfig deserializes across versions.
+
+use std::net::Ipv4Addr;
+
+use bootstore::schemes::v0 as bootstore;
+use omicron_common::api::{
+    external::SwitchLocation,
+    internal::shared::{
+        BgpConfig, BgpPeerConfig, PortConfigV1, PortFec, PortSpeed,
+        RackNetworkConfig, RouteConfig,
+    },
+};
+use omicron_sled_agent::bootstrap::early_networking::{
+    EarlyNetworkConfig, EarlyNetworkConfigBody,
+};
+use omicron_test_utils::dev::test_setup_log;
+
+const BLOB_PATH: &str = "tests/data/early_network_blobs.txt";
+
+/// Test that previous and current versions of `EarlyNetworkConfig` blobs
+/// deserialize correctly.
+#[test]
+fn early_network_blobs_deserialize() {
```

Comment:
FWIW: for little parsers like this, I really like using [nom](https://github.com/rust-bakery/nom).

Reply:
Ah definitely if it's a bit more complex than this. In this case I figured it was just splitting by comma.